clients/hadoopfs/src/main/java/io/lakefs/FileSystemTracer.java

```diff
@@ -0,0 +1,353 @@
+        LOG.trace("create(Path {}, permission {}, overwrite {}, bufferSize {}, replication {}, blockSize {}, progress {})",
+                f, permission, overwrite, bufferSize, replication, blockSize, progress);
+
+        TracerOutputTStream tOutputStream = new TracerOutputTStream(f, permission, overwrite, bufferSize, replication, blockSize, progress);
+        return new FSDataOutputStream(tOutputStream, null);
+    }
+
+    @Override
+    public FSDataOutputStream append(Path f, int bufferSize, Progressable progress) throws IOException {
+        LOG.trace("append(f {}, bufferSize {}, progress {})", f, bufferSize, progress);
+
+        FSDataOutputStream lakefsRes = lfsFileSystem.append(f, bufferSize, progress);
+        FSDataOutputStream s3aRes = s3AFileSystem.append(translateLakeFSPathToS3APath(f), bufferSize, progress);
+        LOG.trace("{}[append] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;//unused
+    }
+
+    @Override
+    public boolean rename(Path src, Path dst) throws IOException {
+        LOG.trace("rename(src {}, dst {})", src, dst);
+
+        boolean lakefsRes = lfsFileSystem.rename(src, dst);
+        boolean s3aRes = s3AFileSystem.rename(translateLakeFSPathToS3APath(src), translateLakeFSPathToS3APath(dst));
+        LOG.trace("{}[rename] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;
+    }
+
+    @Override
+    public boolean delete(Path f, boolean recursive) throws IOException {
+        LOG.trace("delete(f {}, recursive {})", f, recursive);
+
+        boolean lakefsRes = lfsFileSystem.delete(f, recursive);
+        boolean s3aRes = s3AFileSystem.delete(translateLakeFSPathToS3APath(f), recursive);
+        LOG.trace("{}[delete] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+        return s3aRes;
+    }
+
+    @Override
+    public FileStatus[] listStatus(Path f) throws FileNotFoundException, IOException {
+        LOG.trace("listStatus(f {})", f);
+
+        FileStatus[] lakefsRes = lfsFileSystem.listStatus(f);
+        FileStatus[] s3aRes = s3AFileSystem.listStatus(translateLakeFSPathToS3APath(f));
+        LOG.trace("{}[listStatus] lakefs: {}, s3a: {}", RESULTS_COMPARISON, lakefsRes, s3aRes);
+
+        if (useLakeFSFileSystemRes) {
+            return lakefsRes;
+        }
+
+        for (FileStatus stat : s3aRes) {
+            Path s3aPath = stat.getPath();
+            Path lfsPath = translateS3APathToLakeFSPath(s3aPath);
+            stat.setPath(lfsPath);
+        }
+        return s3aRes;
+    }
+
+    /**
+     * Set the current working directory for the given file system. All relative
```

Comment:
note - think we can skip the code doc for the known method as we just trace them and each underlying file system will document the right behaviour better.

Reply:
agreed, I removed the docs from the other methods and missed this one